src/main/java/dinkplugin/SettingsManager.java

```diff
@@ -181,9 +188,50 @@ void onConfigChanged(ConfigChanged event) {
         }
     }
 
+    void onGameState(GameStateChanged event) {
+        if (event.getGameState() != GameState.LOGGED_IN) {
+            justLoggedIn.set(false);
+            return;
+        } else {
+            justLoggedIn.set(true);
+        }
+
+        // Since varbit values default to zero and no VarbitChanged occurs if the
+        // newly received value is equal to the existing value, we must manually
+        // check those where 0 is an invalid value deserving of a warning.
+        clientThread.invokeLater(() -> {
+            if (justLoggedIn.get())
+                return false; // try again on next tick
+
+            if (config.notifyCollectionLog() && client.getVarbitValue(Varbits.COLLECTION_LOG_NOTIFICATION) == 0) {
+                warnForGameSetting(CollectionNotifier.ADDITION_WARNING);
+            }
+
+            if (config.notifyPet() && (client.getVarbitValue(PetNotifier.UNTRADEABLE_LOOT_DROPS) == 0 || client.getVarbitValue(PetNotifier.LOOT_DROP_NOTIFICATIONS) == 0)) {
+                warnForGameSetting(PetNotifier.UNTRADEABLE_WARNING);
+            }
```

Comment:
May clarify the logic

```suggestion
            if (config.notifyCollectionLog() && client.getVarbitValue(Varbits.COLLECTION_LOG_NOTIFICATION) == 0) {
            assert isCollectionLogInvalid(0);
                warnForGameSetting(CollectionNotifier.ADDITION_WARNING);
            }

            if (config.notifyPet() && (client.getVarbitValue(PetNotifier.UNTRADEABLE_LOOT_DROPS) == 0 || client.getVarbitValue(PetNotifier.LOOT_DROP_NOTIFICATIONS) == 0)) {
            assert isPetLootInvalid(0);
                warnForGameSetting(PetNotifier.UNTRADEABLE_WARNING);
            }
```

Reply:
Current code is fine imo